modules/account-lib/src/coin/algo/utils.ts

```diff
@@ -261,44 +255,69 @@ export class Utils implements BaseUtils {
       buffer = decodedTx.msig || decodedTx.sig ? encoding.encode(decodedTx) : encoding.encode(decodedTx.txn);
     }
 
-    if (this.isDecodableUnsignedAlgoTxn(buffer)) {
-      return {
-        rawTransaction: new Uint8Array(buffer),
-        txn: algosdk.decodeUnsignedTransaction(buffer),
-        signed: false,
-      };
-    } else if (this.isDecodableSignedTransaction(buffer)) {
-      // TODO: Replace with
-      // return algosdk.Transaction.from_obj_for_encoding(algosdk.decodeSignedTransaction(buffer).txn);
-      // see: https://github.com/algorand/js-algorand-sdk/issues/364
-      // "...some parts of the codebase treat the output of Transaction.from_obj_for_encoding as EncodedTransaction.
-      // They need to be fixed(or we at least need to make it so Transaction conforms to EncodedTransaction)."
-      const tx = algosdk.decodeSignedTransaction(buffer);
-
-      const signers: string[] = [];
-      const signedBy: string[] = [];
-      if (tx.msig && tx.msig.subsig) {
-        for (const sig of tx.msig.subsig) {
-          const addr = algosdk.encodeAddress(sig.pk);
-          signers.push(addr);
-          if (sig.s) {
-            signedBy.push(addr);
-          }
-        }
-      }
+    try {
+      return this.tryToDecodeUnsignedTransaction(buffer);
+    } catch (_: unknown) {
```

Comment:
https://mariusschulz.com/blog/optional-catch-binding-in-typescript

```suggestion
    } catch {
```